src/tsolvers/axdiffsolver/ArraySolver.h

```diff
@@ -227,8 +227,8 @@ class ArraySolver : public TSolver {
 
     PTRef getEquality(ERef lhs, ERef rhs, Logic & logic) const { return logic.mkEq(egraph.ERefToTerm(lhs), egraph.ERefToTerm(rhs)); }
 
-    bool isFalsified(PTRef equality) const { return this->hasPolarity(equality) and this->getPolarity(equality) == l_False; }
-    bool isSatisfied(PTRef equality) const { return this->hasPolarity(equality) and this->getPolarity(equality) == l_True; }
+    bool isFalsified(PTRef equality) const { return logic.isFalse(equality) or (this->hasPolarity(equality) and this->getPolarity(equality) == l_False); }
```

Comment:
Why do we need this new requirement, if we have `this->getPolarity(equality) == l_False`?
Or is it possible to PTRef to be falsified without polarity?

Reply:
It was not working for the `false` term. The theory solvers currently do not store information about `true` and `false` terms.